### ISSUE TITLE
Feature Request : Github-actions - merge inspect.yml into push.yml 

### ISSUE THREAD

**Issue by aialok** [PalisadoesFoundation/talawa-api#1696]
**Is your feature request related to a problem? Please describe.**

- The current GitHub Actions configuration in the API repository involves two separate YAML files, namely inspect.yml and push.yml, both triggered on every push. To streamline and simplify the workflow, we propose consolidating the code from inspect.yml into push.yml and subsequently removing inspect.yml.

**Describe the solution you'd like**

* In the API repo, the commands in the inspect.yml file are invoked with every push.
* We have a separate push.yml file that is also invoked with every push.
* We need to move the YAML code in the inspect.yml to the push.yml file and delete the inspect.yml file
* When this is done, the name of the job needs to be changed from test to something indicative of doing introspection testing.
* It the migrated YAML code needs to:
be dependent on the Push-Workflow job in the push.yml file
ensure that the inspect.yml file's code automatically adjusts to the branch
* This will mean that developers will only need to edit one file for all push related changes to our GitHub actions

**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**
- This adjustment aims to centralize all push-related changes into a single file (push.yml), making it more straightforward for developers to manage and update GitHub Actions configurations. Additionally, the workflow's naming conventions will be improved to better reflect its purpose.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by aialok** [comment]
I would love to work on this issue : )

**Comment by palisadoes** [comment]
I have to reopen this. The push workflow started to fail at this step. Please investigate

- https://github.com/PalisadoesFoundation/talawa-api/actions/runs/7603568839/job/20705612488

**Comment by aialok** [comment]
Hello @palisadoes,

I reviewed the commit history in the GitHub workflow and identified that the `checkSchema` test is failing after making changes to the `inspect.yml` file in commit [f5b7deaf7eb96b435cf858bff2c62fbfb1e55772](https://github.com/PalisadoesFoundation/talawa-api/commit/f5b7deaf7eb96b435cf858bff2c62fbfb1e55772). The addition of `node_modules` caching for CI workflows seems to be the cause.

Since that commit, the `checkSchema` test has failed consistently with each push.

Additionally, the `generateDocumentation` process is also failing for every push since it was added to the test.

Here is a screenshot of the failure:
![generateDocumentation](https://github.com/PalisadoesFoundation/talawa-api/assets/66772290/8407ef6e-9833-43cd-b614-3b2704555978)
